### DISCUSSION TITLE
Allow set custom event on suggestion click

### DISCUSSION THREAD

**Discussion by SneakyGerald** [digitalfortress-tech/typeahead-standalone#49] [Ideas]
At the moment when we click the suggestions the behavior is to set suggestion text as the input value. However, if we want to run our own or additional events we are not able because of `preventDefault()`:

```
      div.addEventListener('click', function (ev: MouseEvent): void {
        clear();
        selected = item;
        input.value = display(item, ev);
        ev.preventDefault(); // --> this
      });
```
I think it would be useful to allow users to add additional logic on typeahead events and not limit it.

**Comment by niketpathak** [comment]
Hello @SneakyGerald ,

The `display()` callback receives the **MouseEvent | KeyboardEvent | null** and thereby allows you to do whatever you'd like by checking for the existence of the event. 

Another user wanted something similar and here's a link that'll help you add additional logic as per your needs - https://github.com/digitalfortress-tech/typeahead-standalone/discussions/24#discussioncomment-2997561. 
So you could easily use the `display` callback to add your additional logic.

**Comment by SneakyGerald** [comment]
Thanks @niketpathak ! This will work for me, however, if the user wants to add `href` to the suggestion item it will get stopped by `preventDefault()` anyway. We can make a workaround using js in `display()` callback as well but seems too complex compared to simple `<a>` tag solution.

**Reply by niketpathak** [reply]
@SneakyGerald yes, you are right. However that will not give a consistent behavior while selecting a suggestion via the mouse and via the keyboard. 
To elaborate: if we remove `preventDefault()`, a mouse click on a suggestion containing a link will cause the user to navigate to the link. But if the user selects the same suggestion with his keyboard and hits on <kbd>Enter</kbd> or <kbd>Tab</kbd>, he/she will not have the same behaviour. Instead, it will execute the `display()` callback thereby only update the input's value. 

To ensure consistent behavior on selecting a suggestion, we decided to leave `preventDefault()` in place.
Thanks again!

**Reply by niketpathak** [reply]
Hello @SneakyGerald, after reconsideration, you were finally right. The choice to call `preventDefault` has been left upto the user as of [v4.21.0](https://www.npmjs.com/package/typeahead-standalone/v/4.21.0). Thanks for your input.
